### PR TITLE
Phase 2 and Phase 3

### PR DESCRIPTION
**Pull request by rafiulgits**
**Features**
 * Fetch single/paginate/all Post object
 * Content Negotiation : `XML` and `JSON` support for both request and response
 * Update Operation on a single post

### CODE REVIEW

File: Controllers/PostController.cs [hunk 1]
```diff
@@ -27,7 +28,59 @@ public async Task<ActionResult> CreatePost([FromBody] PostDto post)
...
+        [HttpGet("page/{number}")]
+        public async Task<ActionResult<List<Post>>> GetPaginate(int number)
+        {
+            var queryOrder = Request.Query["order"].ToString();
+            if(!String.IsNullOrEmpty(queryOrder))
+            {
+                queryOrder = queryOrder.ToLower();
+                if(queryOrder == "desc")
+                {
+                    return await postService.GetPage(number, 10 ,true);
```
Comment (cefalo-asif) [inline Controllers/PostController.cs:54]: What if our client plans to display 15 records per page?

File: Controllers/PostController.cs [hunk 2]
```diff
@@ -27,7 +28,59 @@ public async Task<ActionResult> CreatePost([FromBody] PostDto post)
...
+        }
+
+        [HttpPut]
+        public async Task<ActionResult<Post>> UpdatePost(PostDto post)
+        {
+            if(post.Id == Guid.Empty)
+            {
+                return BadRequest();
+            }
+            if(!post.IsValid())
```
Comment (cefalo-asif) [inline Controllers/PostController.cs:73]: Two bad request return point? This is one kind of code smell.
Can we somehow validate within the IsValid method as discussed?

File: Data/Dto/PostDto.cs [hunk 3]
```diff
@@ -4,12 +4,12 @@ namespace Blogger.Data.Dto
...
     {
-
+        public Guid Id { set; get; }
         public string Title {set; get;}
         public string Body {set; get;}
         public DateTime CreatedOn {set; get;}
-        
-        public dynamic Errors = null;
+
+        public ErrorDto Error = ErrorDto.Empty();
```
Comment (cefalo-asif) [inline Data/Dto/PostDto.cs:12]: Let's move this to some base class at some point. (Okay for now)

File: README.md [hunk 4]
```diff
@@ -1,17 +1,306 @@
...
-    Url : {Host}/api/Post
-    ```
-    
\ No newline at end of file
+
+
+## Response Status
+
+| Status Code               | Reason                                                       |
+| ------------------------- | ------------------------------------------------------------ |
```
Comment (cefalo-asif) [inline README.md:16]: Good Job!

File: Controllers/PostController.cs (hunk 1 above)
Comment (rafiulgits) [reply Controllers/PostController.cs:54]: then addition query parameter can be passed

File: Controllers/PostController.cs (hunk 2 above)
Comment (rafiulgits) [reply Controllers/PostController.cs:73]: will be update on next phase

**Review by cefalo-asif — APPROVED** [review]

**Pull request merged**